src/metadata_service/fam_metadata_service_direct.cpp

```diff
@@ -755,9 +755,9 @@ void Fam_Metadata_Service_Direct::Impl_::metadata_insert_region(
         if (kvsObj == metadataKvsMap->end()) {
             pthread_rwlock_unlock(&kvsMapLock);
             pthread_rwlock_wrlock(&kvsMapLock);
-            kvsObj = metadataKvsMap->find(regionId);
-            if (kvsObj == metadataKvsMap->end()) {
-                metadataKvsMap->insert({ regionId, kvs });
+            auto res = metadataKvsMap->insert({ regionId, kvs });
+            if (!res.second) {
```

Comment:
Move the delete outside the lock. Also do we need to delete in the else condition at line 763 ?